// render-core/tension/prototype/PrefaceAction.js
import { registerTagLib } from "../../xboot/tagProcessor";
import { set_system_ext_router, set_user_anonymous_ext, set_user_anonymous_field, set_user_ext_http } from "../../system/recorder/table3/system_func_3";
import { set_router_mode } from "../../system/recorder/table1/system_func_1";
var PrefaceAction = /** @class */ (function () {
    function PrefaceAction() {
    }
    PrefaceAction.prototype.add_anonymous_extension = function (name, plugin) {
        set_user_anonymous_ext(name, plugin);
    };
    PrefaceAction.prototype.add_anonymous_filed = function (name, value) {
        set_user_anonymous_field(name, value);
    };
    PrefaceAction.prototype.add_system_router = function (router) {
        set_system_ext_router(router);
    };
    PrefaceAction.prototype.add_user_http = function (http) {
        set_user_ext_http(http);
    };
    PrefaceAction.prototype.register_component = function (component) {
        registerTagLib(component);
    };
    PrefaceAction.prototype.set_field_route_mode = function (value) {
        set_router_mode(value);
    };
    return PrefaceAction;
}());
export { PrefaceAction };
