// render-core/system/recorder/table0/system_func_0.js
import system_t_0 from "./system_t_0";
/**
 * Use the func to inject a tag component to the tag-library.
 * @param tagName The tag name
 * @param tagDefine The tag prototype class
 */
export function tag_library_inject(tagName, tagDefine) {
    if (system_t_0.tagLib.has(tagName))
        console.log("no");
    else
        system_t_0.tagLib.set(tagName, tagDefine);
}
/**
 * Use the func to get a tag prototype class
 * @param tagName The name of tag you want to search
 */
export function tag_library_search(tagName) {
    if (system_t_0.tagLib.has(tagName))
        return system_t_0.tagLib.get(tagName);
    else
        return null;
}
/**
 * From the system_t_0 table, get tag library
 */
export function get_tag_library() {
    return system_t_0.tagLib;
}
/**
 * From the system_t_0 table, get style library
 */
export function get_style_library() {
    return system_t_0.styleLib;
}
/**
 * Get context controller from system_t_0
 */
export function get_context_controller() {
    return system_t_0.contextController;
}
/**
 * Set context controller to system_t_0
 * @param context
 */
export function set_context_controller(context) {
    Reflect.set(system_t_0, "contextController", context);
}
